### PR TITLE
Fix minor issues with docker_login docs

### PR DESCRIPTION
**Pull request by lachlancooper**
##### SUMMARY
The docker_login parameter docs have some minor issues. This fixes some of them.

##### ISSUE TYP
 - Docs Pull Request

##### COMPONENT NAME
docker_login

##### ANSIBLE VERSION
```
ansible 2.4.2.0
```

### CODE REVIEW

**Comment by acozine** [comment]
@lachlancooper thanks for finding and fixing the typo in the docker documentation. What's your thinking about removing the note about Docker Hub on line 49? Has Docker Hub changed its interface? If not, why isn't noting that Docker Hub requires the `email` field (even though Ansible doesn't require it) helpful/useful?

**Comment by lachlancooper** [comment]
@acozine thanks for picking this up. It looks like the `docker login` command used to automatically create a registry account if it didn't exist, which meant that both username and email were required. This was changed in 1.11.0:

https://github.com/moby/moby/pull/20565
https://docs.docker.com/engine/deprecated/#-e-and---email-flags-on-docker-login

Researching this made me realise that I'd missed a few things in this file, so I've added a commit to this PR.

File: lib/ansible/modules/cloud/docker/docker_login.py [hunk 1]
```diff
@@ -316,9 +312,6 @@ def main():
         login_result={}
     )
 
-    if client.module.params['state'] == 'present' and client.module.params['registry_url'] == DEFAULT_DOCKER_REGISTRY and not client.module.params['email']:
```
Comment (gundalow) [inline lib/ansible/modules/cloud/docker/docker_login.py:319]: Why has this changed?

Comment (lachlancooper) [reply lib/ansible/modules/cloud/docker/docker_login.py:319]: Docker Hub (`DEFAULT_DOCKER_REGISTRY`) no longer requires an email address to login, so we don't need to ensure the `email` param is supplied.

**Pull request merged**